src/document/DocumentAnnotator.ts

```diff
@@ -1,21 +1,46 @@
+import debounce from 'lodash/debounce';
 import BaseAnnotator, { Options } from '../common/BaseAnnotator';
 import BaseManager from '../common/BaseManager';
 import { centerRegion, isRegion, RegionManager } from '../region';
 import { Event } from '../@types';
 import { getAnnotation } from '../store/annotations';
-import { Mode } from '../store';
+import { getSelectionItem } from '../highlight/highlightUtil';
+import { Mode, setSelectionAction } from '../store';
 import { scrollToLocation } from '../utils/scroll';
 import './DocumentAnnotator.scss';
 
+const TEXT_LAYER_ENHANCEMENT = 300;
+
 export default class DocumentAnnotator extends BaseAnnotator {
     annotatedEl?: HTMLElement;
 
     managers: Map<number, Set<BaseManager>> = new Map();
 
+    selectionChangeTimer?: number | undefined;
+
     constructor(options: Options) {
         super(options);
 
         this.addListener(Event.ANNOTATIONS_MODE_CHANGE, this.handleChangeMode);
+        document.addEventListener('selectionchange', this.handleSelectionChange);
+    }
+
+    destroy(): void {
+        if (this.selectionChangeTimer) {
+            clearTimeout(this.selectionChangeTimer);
+        }
+
+        this.removeListener(Event.ANNOTATIONS_MODE_CHANGE, this.handleChangeMode);
```

Comment:
Is this a bugfix from a prior commit?

Reply:
Yeah, when I implemented the `destroy()`, I found the prior commit forgot to remove the listener. It's just one line so I don't want to open a PR for it.